File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-typescript/fabricas/FabricaWolks.ts
import { Carro } from '../carros/Carro'
import { Gol } from '../carros/Gol'
import { FabricaDeCarro } from './FabricaDeCarro'

export class FabricaWolks extends FabricaDeCarro {

	public criarCarro(): Carro {
		return new Gol();
	}

}

File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-java/src/br/edu/ifpb/padroes/diegofrazao/fabricas/FabricaWolks.java
package br.edu.ifpb.padroes.diegofrazao.fabricas;

import br.edu.ifpb.padroes.diegofrazao.carros.Carro;
import br.edu.ifpb.padroes.diegofrazao.carros.Gol;

public class FabricaWolks implements FabricaDeCarro {
    @Override
    public Carro criarCarro() {
        return new Gol();
    }
}
